js/jobrow.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {Link, hashHistory} from 'react-router';
+
+import {
+  timeFormat,
+  secondFormat,
+  jobState,
+} from './utils';
+import ProgressBar from './components/progress-bar';
+
+// Lifted from react-router.
+const isLeftClickEvent = (e) => e.button === 0;
+const isModifiedEvent = (event) => event.metaKey || event.altKey || event.ctrlKey || event.shiftKey;
+
+class JobRow extends React.Component {
+  constructor() {
+    super();
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(e) {
+    if (isLeftClickEvent(e) && !isModifiedEvent(e)) {
+      hashHistory.push(`/job/${this.props.job.id}`);
+    }
+  }
+
+  render() {
+    const columns = this.columns();
```

Comment:
where is `columns` defined..?

Reply:
Oh I see, it's implemented in a subclass

Reply:
🙈

Reply:
i could add a default implementation that throws an exception…